great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3948,15 +3952,17 @@ def get_available_data_asset_names(
             for datasource_name in datasource_names:
                 try:
                     datasource = self.get_datasource(datasource_name)
-                    assert not isinstance(
-                        datasource, FluentDatasource
-                    ), 'Method "get_available_data_asset_names" not implemented for FluentDatasource'
-                    data_asset_names[
-                        datasource_name
-                    ] = datasource.get_available_data_asset_names()
+                    if isinstance(datasource, FluentDatasource):
+                        data_asset_names[datasource_name] = list(
+                            datasource.get_asset_names()
+                        )
+                    else:
+                        data_asset_names[datasource_name] = list(
+                            datasource.get_available_data_asset_names()
+                        )
                 except ValueError:
                     # handle the edge case of a non-existent datasource
-                    data_asset_names[datasource_name] = {}
+                    data_asset_names[datasource_name] = list()
```

Comment:
Nit - can we use `[]`? Either works but we tend to use this syntax in our codebases